isv/src/prediction.py

```diff
@@ -21,9 +22,18 @@ class ACMGClassification(enum.StrEnum):
     BENIGN = "Benign"
 
 
-def get_shap_values(loaded_model: Any, input_df: pd.DataFrame) -> dict[str, float]:
-    explainer_cnvs = shap.Explainer(loaded_model)
-    shap_values = explainer_cnvs(input_df).values[0]
+def get_shap_values(loaded_model: Any, input_df: pd.DataFrame, cnv_type: str) -> dict[str, float]:
+    path_to_train_set = os.path.join(core.MODELS_DIR, f"X_train_clinvar_{cnv_type}.tsv.gz")
+    X_train = pd.read_csv(path_to_train_set, sep='\t', compression='gzip')
+
+    attributes = get_attributes(cnv_type)
+    X_train = X_train[attributes]
+    scaler = RobustScaler()
+    X_train = scaler.fit_transform(X_train)
+    X_any = scaler.transform(input_df)
+
+    explainer_cnvs = shap.TreeExplainer(loaded_model, X_train, model_output='probability')
```

Comment:
toto sa neda nejako prerobit tak aby sa ulozil uz samotny explainer? Aby sa to stale nemuselo robit pre kazde volanie programu?

Reply:
Prerobene. je ulozeny aj explainer aj scaler.  trenovacie datasety su vymazane.